cylc/flow/task_pool.py

```diff
@@ -245,15 +245,15 @@ def release_runahead_tasks(self):
         # At restart all tasks are runahead-limited but finished and manually
         # triggered tasks (incl. --start-task's) can be released immediately.
         for itask in (
-            itask
-            for itask in self.get_tasks()
-            if itask.state.is_runahead
-            if itask.state(
+            jtask
+            for jtask in self.get_tasks()
+            if jtask.state.is_runahead
+            if jtask.state(
                 TASK_STATUS_FAILED,
                 TASK_STATUS_SUCCEEDED,
                 TASK_STATUS_EXPIRED
             )
-            or itask.is_manual_submit
+            or jtask.is_manual_submit
         ):
             self.release_runahead_task(itask)
             released = True
```

Comment:
As above
```diff
diff --git a/cylc/flow/task_pool.py b/cylc/flow/task_pool.py
index 20d2cd98b..f180e0283 100644
--- a/cylc/flow/task_pool.py
+++ b/cylc/flow/task_pool.py
@@ -244,19 +244,17 @@ class TaskPool:
 
         # At restart all tasks are runahead-limited but finished and manually
         # triggered tasks (incl. --start-task's) can be released immediately.
-        for itask in (
-            itask
-            for itask in self.get_tasks()
-            if itask.state.is_runahead
-            if itask.state(
-                TASK_STATUS_FAILED,
-                TASK_STATUS_SUCCEEDED,
-                TASK_STATUS_EXPIRED
-            )
-            or itask.is_manual_submit
-        ):
-            self.release_runahead_task(itask)
-            released = True
+        for itask in self.get_tasks():
+            if itask.state.is_runahead and (
+                itask.state(
+                    TASK_STATUS_FAILED,
+                    TASK_STATUS_SUCCEEDED,
+                    TASK_STATUS_EXPIRED
+                )
+                or itask.is_manual_submit
+            ):
+                self.release_runahead_task(itask)
+                released = True
 
         runahead_limit_point = self.compute_runahead()
         if not runahead_limit_point:
```

Reply:
Oh, unless the same thing as written below applies?  https://github.com/cylc/cylc-flow/blob/4ca97df6b05e3670dae42ca4d5e680912d0cfdbc/cylc/flow/task_pool.py#L265-L269

in which case using an intermediate list would be fine (also clearer than the current code)